tests/cfg.c

```diff
@@ -0,0 +1,266 @@
+#undef NDEBUG
+
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <assert.h>
+
+#include <security/pam_appl.h>
+
+#include "cfg.h"
+
+static void write_conf(int fd, const char *text) {
+  size_t len;
+  int e;
+
+  len = strlen(text);
+  while (len) {
+    ssize_t w;
+
+    w = write(fd, text, len);
+    assert(w >= 0);
+
+    len -= w;
+    text += w;
+  }
+
+  e = fsync(fd);
+  assert(e == 0);
+}
+
+static char *generate_conf_file_arg(void) {
+  // Generate a conf_file= argument
+  //
+  // The function returns a string which is:
+  // - suitable as argv item for pam_u2f
+  // - suitable as template argument for mkstemp
+  // - referring to the absolute path of the temporary file.
+
+  char cwd[1024];
```

Comment:
Should we just use `PATH_MAX`?

Reply:
`PATH_MAX` is not portable, so I would go for  `fpathconf(dirfd, _PC_PATH_MAX)`.
The implication is of course `malloc`+`free`, but that's acceptable to me.

Reply:
Well, 4096 then.